xapian-core/weight/tfidfweight.cc

```diff
@@ -136,7 +142,8 @@ TfIdfWeight::get_maxpart() const
 {
     Xapian::termcount wdf_max = get_wdf_upper_bound();
     Xapian::termcount len_min = get_doclength_lower_bound();
-    double wdfn = get_wdfn(wdf_max, len_min, normalizations[0]);
+    Xapian::termcount len_max = get_doclength_upper_bound();
+    double wdfn = get_wdfn(wdf_max, len_min, len_max, normalizations[0]);
     return get_wtn(wdfn * idfn, normalizations[2]) * wqf_factor;
```

Comment:
I understand what you're saying. I'm wondering if there is a simpler way of pulling `uniqterms` stat here?

As of now I can't see how to pull `uniqterms` stat in this `double get_maxpart()` method since it lacks the required parameters. So, we'd need to change its declaration in `weight.h` to provide `Xapian::termcount uniqterms` parameters which would further require changing its definition in all the weighting schemes subclasses.